### PR TITLE
refact(errors) - Wired the ingest errors to notifications and activity stream

### PR DESCRIPTION
**Pull request by akintolga**
Major changes in this PR:
* Administrators will get every caught ingest error in Notifications and activity streams (and email)
* Notifying user in the messages will be "System"

So after the merge the following existing features will work as expected:
* The notifications can be turned on/off per Provider on settings
* Administrators can opt out from ingest error emails in user settings

### CODE REVIEW

File: superdesk/io/ftp.py [hunk 1]
```diff
@@ -50,30 +51,42 @@ def _update(self, provider):
...
-                        ftp.retrbinary('RETR %s' % filename, f.write)
-                except FileExistsError:
-                    continue
+                    dest = '%s/%s' % (config['dest_path'], filename)
 
-                xml = etree.parse(dest).getroot()
-                items.append(get_xml_parser(xml).parse_message(xml))
-        return items
+                    try:
+                        with open(dest, 'wb') as f:
```
Comment (petrjasek) [inline superdesk/io/ftp.py]: the 'x' there was important - it makes sure there is only 1 file created - and thus checks if this file was not processed already

**Comment by mugurrus** [comment]
:+1:

File: superdesk/io/ftp.py (hunk 1 above)
Comment (akintolga) [inline superdesk/io/ftp.py]: I see, but I couldn't find any reference to 'x' in python documentation. I wonder if it is a C like feature. I referred to this page: 
https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files
and it says:
> 'w' for only writing (an existing file with the same name will be erased)

Comment (petrjasek) [inline superdesk/io/ftp.py]: you can find it in [open docs](https://docs.python.org/3/library/functions.html#open)

Comment (akintolga) [inline superdesk/io/ftp.py]: Fair enough. I'll revert it.

**Comment by petrjasek** [comment]
:+1:

**Comment by ioanpocol** [comment]
:+1:

**Comment by amagdas** [comment]
:+1:

**Pull request merged**